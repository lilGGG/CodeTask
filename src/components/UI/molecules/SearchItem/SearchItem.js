import SearchInput from "../../atoms/SearchInput/SearchInput";
import Label from "../../atoms/Label/Label";

/**
 * Component for searching input with its label
 * @param {string} text 
 * @param {Method} inputHandler 
 * @returns 
 */
function SearchItem({
  text,inputHandler
}) {
  return (
    <div> 
           <Label Text={text}/>
           <SearchInput inputHandler={inputHandler} />
    </div>
  );
}

export default SearchItem;
