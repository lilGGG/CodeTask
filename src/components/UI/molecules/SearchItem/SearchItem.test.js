import React from "react";
import SearchItem from "./SearchItem";
import ReactDom from "react-dom";
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
// import { create, act } from "react-test-renderer";
// import { shallow, mount, render } from "enzyme";

describe('SearchItem test', () => {
    it('renders snapshot', () => {
      const tree = renderer.create(<SearchItem text="texta" inputHandler={()=>{}}
       />).toJSON();
      expect(tree).toMatchSnapshot();
    });

  });
  
