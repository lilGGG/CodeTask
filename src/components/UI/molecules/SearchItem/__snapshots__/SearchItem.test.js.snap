// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem test renders snapshot 1`] = `
<div>
  <label
    className="sc-gsnTZi jlUXq"
  >
    texta
  </label>
  <input
    className="sc-bczRLJ lcYJBc"
    onChange={[Function]}
    placeholder="Enter Tv Show Title"
  />
</div>
`;
